helm/cluster-vsphere/values.schema.json

```diff
@@ -121,8 +121,34 @@
             "title": "Cluster",
             "description": "Helm values for the provider-independent cluster chart.",
             "default": {
+                "internal": {
+                    "advancedConfiguration": {
+                        "controlPlane": {
+                            "apiServer": {
+                                "extraArgs": {
+                                    "requestheader-allowed-names": "front-proxy-client"
+                                }
+                            }
+                        }
+                    }
+                },
                 "providerIntegration": {
                     "controlPlane": {
+                        "kubeadmConfig": {
+                            "files": [
+                                {
+                                    "contentFrom": {
+                                        "secret": {
+                                            "key": "content",
+                                            "name": "kubevip-pod",
```

Comment:
Out of curiosity what is kube vip?

Reply:
it's our poor man's loadbalancer. vmware customers have to license different bits of their vsphere cluster and some choose to integrate vsphere directly with their upstream networking gear instead of paying the fee for the fancy vsphere networking. consequently, there is no reference network architecture for vsphere and so the upstream chart uses kube-vip to provide floating IPs for loadbalancers - we use it for the api and for any ingress too. it uses ARP underneath so it will work with any customer environment.

Reply:
🤯 wow TIL, thanks!